internal/kafka/internal/services/cloud_providers_test.go

```diff
@@ -501,3 +555,113 @@ func Test_ListCloudProviderRegions(t *testing.T) {
+		fields  fields
+		args    args
+		want    []api.CloudRegion
+		setupFn func()
+		wantErr *Error.ServiceError
+	}{
+		{
+			name: "successful get cloud provider regions from various provider types",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory: &clusters.ProviderFactoryMock{
+					GetProviderFunc: func(providerType api.ClusterProviderType) (clusters.Provider, error) {
+						return &clusters.ProviderMock{
+							GetCloudProvidersFunc: func() (*types.CloudProviderInfoList, error) {
+								return &types.CloudProviderInfoList{
+									Items: []types.CloudProviderInfo{
+										{
+											ID:          "azure",
+											Name:        "azure",
+											DisplayName: "azure",
+										},
+										{
+											ID:          "aws",
+											Name:        "aws",
+											DisplayName: "aws",
+										},
+									},
+								}, nil
+							},
+							GetCloudProviderRegionsFunc: func(providerInf types.CloudProviderInfo) (*types.CloudProviderRegionInfoList, error) {
+								return &types.CloudProviderRegionInfoList{
+									Items: []types.CloudProviderRegionInfo{
+										{
+											ID:              "af-east-1",
+											CloudProviderID: providerInf.ID,
+											Name:            "af-east-1",
+											DisplayName:     "af-east-1",
+										},
+										{
+											ID:              "af-south-1",
+											CloudProviderID: providerInf.ID,
+											Name:            "af-south-1",
+											DisplayName:     "af-south-1",
+										},
+									},
+								}, nil
+							},
+						}, nil
+					}},
+				cache: cache.New(5*time.Minute, 10*time.Minute),
+			},
+			setupFn: func() {
+				mocket.Catcher.Reset()
+				mocket.Catcher.NewMock().
+					WithQuery("SELECT DISTINCT").
+					WithReply([]map[string]interface{}{{"provider_type": "ocm"}, {"provider_type": "standalone"}})
+			},
+			args: args{
+				id: "azure",
+			},
+			want: []api.CloudRegion{
+				{
+					Kind:                   "",
+					Id:                     "af-east-1",
+					DisplayName:            "af-east-1",
+					CloudProvider:          "azure",
+					Enabled:                false,
+					SupportedInstanceTypes: nil,
+					Capacity:               nil,
+				},
+				{
+					Kind:                   "",
+					Id:                     "af-south-1",
+					DisplayName:            "af-south-1",
+					CloudProvider:          "azure",
+					Enabled:                false,
+					SupportedInstanceTypes: nil,
+					Capacity:               nil,
+				},
+			},
+			wantErr: nil,
+		},
+	}
+	RegisterTestingT(t)
+	for _, tt := range tests {
+		tt.setupFn()
+		t.Run(tt.name, func(t *testing.T) {
+			p := cloudProvidersService{
+				providerFactory:   tt.fields.providerFactory,
+				connectionFactory: tt.fields.connectionFactory,
+				cache:             tt.fields.cache,
+			}
+			got, err := p.ListCachedCloudProviderRegions(tt.args.id)
+			Expect(got).To(Equal(tt.want))
+			Expect(err).To(Equal(tt.wantErr))
```

Comment:
This comparison is not correct.

`wantErr` defines whether an error is expected or not (this is, a boolean) , whereas `err` is a `*errors.ServiceError` (which is an implementation of a Go `error`).
The check that needs to be done here is whether there's been an error received or not

Reply:
Could you provide an example of what you mean by this?
`The check that needs to be done here is whether an error has been received or not, not the value of the error itself`

Reply:
Yes.

When `ListCachedCloudProviderRegions` works correctly the returned error is not set (it is `nil`). When it fails it returns an error with a non-nil value.

You want to check if the error has been set or not and not the value of the error itself.
The current ocde is comparing a boolean with an error which is incorrect.

Reply:
Ah thanks for clearing that up :)